lib/prawn/document.rb

```diff
@@ -384,7 +384,7 @@ def render(*a, &b)
     #   pdf.render_file "foo.pdf"
     #
     def render_file(filename)
-      File.open(filename, 'wb') { |f| render(f) }
+      File.open(filename, 'rb+') { |f| render(f) }
```

Comment:
A file opened with 'wb' does not allow reading. My seek-solution reads the rendered body to avoid a second render pass. As you already pointed pointed out the flaw with seeking, I will revert this change.